src/main/java/Ui.java

```diff
@@ -0,0 +1,30 @@
+public class Ui {
+    public void printLogo() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+    }
+
+    public void greetDuke() {
+        System.out.println("Hey there, Red. Anything I can do for you?");
+    }
+
+    public void listTasks(TaskList taskList) {
+        taskList.tasks.forEach(task -> System.out.println(String.format(
+                "%d. %s",
+                (taskList.tasks.indexOf(task) + 1),
+                task.toString())));
```

Comment:
Nice use of functional programming here!

Reply:
Thank you!